code/datums/minimap.dm

```diff
@@ -0,0 +1,415 @@
+	var/announced_start_delay = FALSE
+
+/datum/minimap_section
+	var/x
+	var/y
+	var/z
+	var/width
+	var/height
+	var/png_path
+
+/datum/minimap_data
+	var/z_level
+	var/save_location
+
+	var/tile_offset_x = 0 //! World offset for the lowest x value of the tiles
+	var/tile_offset_y = 0 //! World offset for the lowest y value of the tiles
+	var/tile_width = 0 //! Total number of tiles in the x direction
+	var/tile_height = 0 //! Total number of tiles in the y direction
+	var/section_width = 0 //! Number of tiles in a section in the x direction
+	var/section_height = 0 //! Number of tiles in a section in the y direction
+	var/section_columns = 0 //! Number of sections in the x direction
+	var/section_rows = 0 //! Number of sections in the y direction
+
+	var/datum/asset/simple/minimap_asset //! Asset of the minimap, registered on finalization
+
+/datum/minimap_data/proc/save_persistence_data()
+	text2file(json_encode(list(
+		"map_config_key" = SSmapping.config.config_filename,
+		"git_commit" = GLOB.revdata.commit,
+		"z_level" = z_level,
+		"save_location" = save_location,
+		"tile_offset_x" = tile_offset_x,
+		"tile_offset_y" = tile_offset_y,
+		"tile_width" = tile_width,
+		"tile_height" = tile_height,
+		"section_width" = section_width,
+		"section_height" = section_height,
+		"section_columns" = section_columns,
+		"section_rows" = section_rows,
+		"asset_paths" = json_encode(minimap_asset.assets),
+	)), "[save_location]/persistence.json")
+
+/datum/minimap_data/proc/try_load_persistence_data()
+	var/persistence_raw = file2text("[save_location]/persistence.json")
+	if(!length(persistence_raw))
+		return FALSE
+
+	var/list/persistence_data = json_decode(persistence_raw)
+	if(!length(persistence_data))
+		return FALSE
+	if(persistence_data["map_config_key"] != SSmapping.config.config_filename)
+		return FALSE
+	if(persistence_data["save_location"] != save_location)
+		return FALSE
+	if(persistence_data["git_commit"] != GLOB.revdata.commit)
+		return FALSE
+
+	z_level = persistence_data["z_level"]
+	save_location = persistence_data["save_location"]
+	tile_offset_x = persistence_data["tile_offset_x"]
+	tile_offset_y = persistence_data["tile_offset_y"]
+	tile_width = persistence_data["tile_width"]
+	tile_height = persistence_data["tile_height"]
+	section_width = persistence_data["section_width"]
+	section_height = persistence_data["section_height"]
+	section_columns = persistence_data["section_columns"]
+	section_rows = persistence_data["section_rows"]
+	minimap_asset = new /datum/asset/simple
+	minimap_asset.assets = json_decode(persistence_data["asset_paths"])
+	minimap_asset.register()
+	return TRUE
+
+/// Holds information used for generating a minimap.
+/datum/minimap_generation_cache
+	var/started_at //! When the generation of the minimap was queued
+	var/list/datum/minimap_generation_section/sections
+	var/datum/minimap_generation_section/current_section
+
+	var/datum/minimap_data/data //! The data for the minimap
```

Comment:
why are you only commenting some of these?
also while //! SHOULD work it doesn't currently, is there a reason you're using it?

Reply:
the ones that aren't commented have self evident names. for example `current_section` is pretty clear